src/JanusStream.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file JanusStream.h
+ * @author Daniel Stiner (danstiner@gmail.com)
+ * @date 2021-03-10
+ * @copyright Copyright (c) 2021 Daniel Stiner
+ */
+
+#include "JanusStream.h"
+
+#pragma region Constructor/Destructor
+JanusStream::JanusStream(
+    ftl_channel_id_t channelId,
+    ftl_stream_id_t streamId,
+    MediaMetadata mediaMetadata) :
+    channelId(channelId),
+    streamId(streamId),
+    mediaMetadata(mediaMetadata)
+{ }
+#pragma endregion
+
+#pragma region Public methods
+void JanusStream::SendRtpPacket(const std::vector<std::byte>& packet)
+{
+    std::lock_guard lock(mutex);
```

Comment:
Can we do a shared_lock here since we aren't writing any new data?

Reply:
I haven't profiled to prove it but my understand is for cases like this where there is little to no reader contention it is more efficient to use a standard mutex.

Basically the transport thread is locking this 99.99% of the time for just the few microseconds it takes to put the packet on these destination queues. The janusftl thread is only locking occasionally to add/remove destinations. So a shared lock isn't really any benefit and in my understanding is actually slower.

Reply:
That makes sense! TIL